implementation/map-to-scatter-gather/scatter_gather/lambda_/requester/app.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+# -*- coding:utf-8 -*-
+###
+# File: app.py
+# Created Date: Thursday, March 16th 2023, 4:53:36 pm
+# Author: Agostino Di Figlia
+# -----
+# Copyright (c) 2023 Amazon Web Services
+# 
+# 2022 Amazon Web Services, Inc. or its affiliates. All Rights Reserved.
+# This AWS Content is provided subject to the terms of the AWS Customer Agreement available at  
+# http://aws.amazon.com/agreement or other written agreement between Customer and either
+# Amazon Web Services, Inc. or Amazon Web Services EMEA SARL or both.
+# Note:
+# THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+###
+import json
+import boto3
+import os
+import sys
+import logging
+import uuid
+
+logging.basicConfig(format='%(asctime)s %(levelname)-8s %(message)s', level=logging.INFO, datefmt='%Y-%m-%d %H:%M:%S')
+logging.getLogger().setLevel(logging.INFO)
+
+client_sns = boto3.client('sns')
+
+MAX_SCATTER = os.getenv('MAX_SCATTER')
+
+def lambda_handler(event, context):
+    
+    uuid_quote = uuid.uuid4()
+    # assumption data is present in event
+    event['data']['uuid'] = str(uuid_quote)
+    message = event
+    
+    if MAX_SCATTER is not None:
+        message = []
+        for n in range(0,int(MAX_SCATTER)):
```

Comment:
This code assumes that all requests are in a single message? That might not be the case for a general scenario.